src/transaction.ts

```diff
@@ -435,6 +421,54 @@ export class Transaction {
 
     return feeForMove.plus(processingFee);
   }
+
+  /**
+   * Creates a new Transaction object from a DraftTransaction.
+   */
+  static fromDraft(draft: DraftTransaction): Transaction {
+    return new Transaction({
+      sender: Address.fromBech32(draft.sender),
+      receiver: Address.fromBech32(draft.receiver),
+      gasLimit: new BigNumber(draft.gasLimit).toNumber(),
+      chainID: "",
+      value: draft.value,
+      data: new TransactionPayload(Buffer.from(draft.data))
+    })
+  }
+
+  /**
+   * Creates a new Transaction object from a TransactionNext object.
+   */
+  static fromTransactionNext(transaction: ITransactionNext): Transaction {
+    const tx =  new Transaction({
+      sender: Address.fromBech32(transaction.sender),
+      receiver: Address.fromBech32(transaction.receiver),
+      gasLimit: new BigNumber(transaction.gasLimit).toNumber(),
+      chainID: transaction.chainID,
+      value: new BigNumber(transaction.value).toFixed(0),
+      data: new TransactionPayload(Buffer.from(transaction.data)),
+      nonce: Number(transaction.nonce),
+      gasPrice: Number(transaction.gasPrice),
+      receiverUsername: transaction.receiverUsername,
+      senderUsername: transaction.senderUsername,
+      options: transaction.options,
+      version: transaction.version
+    });
+
+    if (transaction.guardian) {
+      tx.guardian = Address.fromBech32(transaction.guardian)
+    }
+
+    if (transaction.signature.length) {
+      tx.applySignature(transaction.signature);
+    }
+    
+    if (transaction.guardianSignature) {
```

Comment:
The checks for `transaction.signature` and `transaction.guardianSignature` are not symmetric.

Reply:
Indeed, they were not. Fixed.